ActionScript/ActionScript.sublime-syntax

```diff
@@ -12,87 +12,892 @@ first_line_match: |-
-          pop: true
-        - match: ^\s*(\*)(?!/)
-          captures:
-            1: punctuation.definition.comment.actionscript.2
-    - match: (//).*$\n?
-      scope: comment.line.double-slash.actionscript.2
-      captures:
-        1: punctuation.definition.comment.actionscript.2
-    - match: \b(instanceof)\b
-      scope: keyword.operator.actionscript.2
-    - match: "[-!%&*+=/?:]"
-      scope: keyword.operator.symbolic.actionscript.2
+    - match: \b(?:{{classes}})\b
+      scope: support.class.actionscript
+
+    - match: \b(?:{{functions}})\b
+      scope: support.function.actionscript
+
+    - match: \b(?:{{constants}})\b
+      scope: support.constant.actionscript
+
+    - match: \b(?:{{keywords}})\b
+      scope: keyword.control.actionscript
+
+    - match: \b(?:{{types}})\b
+      scope: storage.type.actionscript
+
+    - include: constants
+
+    - include: strings
+
+    - match: \b(?:instanceof)\b
+      scope: keyword.operator.actionscript
+
+    - match: '[-!%&*+=/?:]'
+      scope: keyword.operator.symbolic.actionscript
+
     - match: '^[ \t]*(#)[a-zA-Z]+'
-      scope: meta.preprocessor.actionscript.2
+      scope: meta.preprocessor.actionscript
       captures:
-        1: punctuation.definition.preprocessor.actionscript.2
+        1: punctuation.definition.preprocessor.actionscript
+
     - match: '\b(function)\s+([a-zA-Z_]\w*)\s*(\()'
       captures:
-        1: keyword.declaration.function.actionscript.2
-        2: entity.name.function.actionscript.2
-        3: punctuation.definition.parameters.begin.actionscript.2
+        1: keyword.declaration.function.actionscript
+        2: entity.name.function.actionscript
+        3: punctuation.definition.parameters.begin.actionscript
       push:
-        - meta_scope: meta.function.actionscript.2
+        - meta_scope: meta.function.actionscript
+
         - match: \)
-          captures:
-            0: punctuation.definition.parameters.end.actionscript.2
-          pop: true
+          scope: punctuation.definition.parameters.end.actionscript
+          pop: 1
+
         - match: '[^,)\n]+'
-          scope: variable.parameter.function.actionscript.2
+          scope: variable.parameter.function.actionscript
+
     - match: '\b(class)\s+([a-zA-Z_](?:\w|\.)*)(?:\s+(extends)\s+([a-zA-Z_](?:\w|\.)*))?'
-      scope: meta.class.actionscript.2
+      scope: meta.class.actionscript
+      captures:
+        1: keyword.declaration.class.actionscript
+        2: entity.name.type.class.actionscript
+        3: storage.modifier.extends.actionscript
+        4: entity.other.inherited-class.actionscript
+
+  prototype:
+    - include: comments
+
+##[ Comments ]##########################################################################################################
+
+  comments:
+    - include: line-comments
+    - include: block-comments
+
+  block-comments:
+    - include: block-comment-basic
+
+  block-comment-basic:
```

Comment:
Do you expect other kinds of block comments? Otherwise distinction between block and line would be probably enough.